api/v1alpha1/k8s_types.go

```diff
@@ -29,7 +28,7 @@ type ImageSpec struct {
 	// Defaults to Always if :latest tag is specified, or IfNotPresent otherwise.
 	// +kubebuilder:validation:Enum=Always;Never;IfNotPresent
 	// +optional
-	PullPolicy string `json:"pullPolicy,omitempty"`
+	PullPolicy corev1.PullPolicy `json:"pullPolicy,omitempty"`
```

Comment:
Last time I tried, Kubebuilder didn't support enum validation for any other types but strings. Did you check if the validation works?

Reply:
of course:
```
$ git diff
diff --git a/config/samples/perf_v1alpha1_iperf3.yaml b/config/samples/perf_v1alpha1_iperf3.yaml
index 06078e7..20a8b02 100644
--- a/config/samples/perf_v1alpha1_iperf3.yaml
+++ b/config/samples/perf_v1alpha1_iperf3.yaml
@@ -6,7 +6,7 @@ metadata:
 spec:
   image:
     name: xridge/iperf3:3.7.0
-    pullPolicy: IfNotPresent
+    pullPolicy: Always
     # pullSecret: null
 
   serverConfiguration:
```

Result:
```
$ k get deployments -o yaml | grep -i imagePullPolicy
          imagePullPolicy: Always
```

Reply:
Please try an invalid value (*not* Always, Never or IfNotPresent!) and see if the validation still works!

Reply:
kubebuilder currently does not create enums for validations specified in variable. 
workaround is to create the validation for a new type (`type PullPolicy string` on package scope).